File: ruby/two_sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  hash = Hash.new
  nums.each_with_index {|n,i| return [hash[n]-1,i] if hash[n] != nil; hash[target-n] = i + 1 }
end

# 通过减法和index来查找目标值

File: python/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for index, value in enumerate(nums):
            d[value] = index
        for index1, value in enumerate(nums):
            if (target - value) in d:
                index2 = d[target - value]
                if index1 != index2:
                    return [index1, index2]
